corehq/form_processor/migrations/0092_auto_20200924_1753.py

```diff
@@ -0,0 +1,21 @@
+# Generated by Django 2.2.13 on 2020-09-24 17:53
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('form_processor', '0091_auto_20190603_2023'),
+    ]
+
+    operations = [
+        migrations.RemoveIndex(
+            model_name='casetransaction',
+            name='fp_casetrans_formid_05e599_idx',
+        ),
+        migrations.AddIndex(
+            model_name='casetransaction',
+            index=models.Index(fields=['form_id'], name='form_proces_form_id_f2403a_idx'),
+        ),
```

Comment:
@snopoke I know we've set things up on the form and case tables very carefully including possibly custom indexes. Is it surprising to you that this auto migration gets created? I don't have quite enough context to understand it but it seems suspicious

Reply:
@dannyroberts this was surprising to me but when I ran `sqlmigrate` it came out as a no-op. I don't fully understand why.

Reply:
I was wrong - this is not a no-op (I ran sqlmigrate against the wrong DB on my local machine):
```
--
-- Remove index fp_casetrans_formid_05e599_idx from casetransaction
--
DROP INDEX IF EXISTS "fp_casetrans_formid_05e599_idx";
--
-- Create index form_proces_form_id_f2403a_idx on field(s) form_id of model casetransaction
--
CREATE INDEX "form_proces_form_id_f2403a_idx" ON "form_processor_casetransaction" ("form_id");
```

I think we can change this to a rename:
```python
OLD_NAME = "fp_casetrans_formid_05e599_idx"
NEW_NAME = "form_proces_form_id_f2403a_idx"

class Migration(migrations.Migration):

    dependencies = [
        ('form_processor', '0091_auto_20190603_2023'),
    ]

    operations = [
        migrations.RunSQL(
            f"ALTER INDEX {OLD_NAME} RENAME TO {NEW_NAME}",
            f"ALTER INDEX {NEW_NAME} RENAME TO {OLD_NAME}",
            state_operations=[
                migrations.RemoveIndex(
                    model_name='casetransaction',
                    name='fp_casetrans_formid_05e599_idx',
                ),
                migrations.AddIndex(
                    model_name='casetransaction',
                    index=models.Index(fields=['form_id'], name='form_proces_form_id_f2403a_idx'),
                ),
            ]
        )
    ]
```

Reply:
@snopoke Do you have any guess why this migration was created? I took a look at changes to `CaseTransaction` and don't see any index changes since @mkangia created `fp_casetrans_formid_05e599_idx` in https://github.com/dimagi/commcare-hq/commit/f6002d073b058e781803b61b39d6e23ec98332f1

Reply:
I think the initial name did not match Django's generated name so Django did not consider this index to match the one defined on the model.

If you look at the other index that was created at the same time you'll see that it uses a different naming scheme: https://github.com/dimagi/commcare-hq/blob/30eef3f76e964b888c5f45cab600980b01e90df6/corehq/form_processor/migrations/0091_auto_20190603_2023.py#L7